transforms/universal/fdedup/ray/src/data_cleaning_transform_ray.py

```diff
@@ -0,0 +1,138 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import os
+from typing import Any
+
+import ray
+from data_cleaning_transform import (
+    DataCleaningTransform,
+    DataCleaningTransformConfiguration,
+    dataclean_data_access_key,
+    dataclean_data_factory_key,
+    duplicate_list_location_default,
+    duplicate_list_location_key,
+)
+from data_processing.data_access import DataAccessFactoryBase
+from data_processing.utils import CLIArgumentProvider, get_logger
+from data_processing_ray.runtime.ray import (
+    DefaultRayTransformRuntime,
+    RayTransformLauncher,
+)
+from data_processing_ray.runtime.ray.runtime_configuration import (
+    RayTransformRuntimeConfiguration,
+)
+from ray.actor import ActorHandle
+
+
+logger = get_logger(__name__)
+
+
+class DataCleaningRayTransform(DataCleaningTransform):
+    """ """
+
+    def __init__(self, config: dict):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments defined
+        by the companion runtime, LangSelectorTransformRuntime.  If running inside the RayMutatingDriver,
+        these will be provided by that class with help from the RayMutatingDriver.
+        """
+        docs2removedf = config.get("df", None)
+        if docs2removedf is not None:
+            # This is recommended for production approach. In this case domain list is build by the
+            # runtime once, loaded to the object store and can be accessed by actors without additional reads
+            try:
+                config["df"] = ray.get(config.get("df"))
+            except Exception as e:
+                self.logger.warning(f"Exception loading docs2remove list from ray object storage {e}")
+                raise RuntimeError(f"exception loading from object storage for key {docs2removedf}")
+        super().__init__(config)
+
+
+class DataCleaningRuntime(DefaultRayTransformRuntime):
+    """
+    Ingest Data cleaning runtime support
+    """
+
+    def __init__(self, params: dict[str, Any]):
+        """
+        Create filter runtime
+        :param params: parameters, that should include
+            ingest_supported_langs_file_key: supported languages file
+            ingest_detect_programming_lang_key: whether to detect programming language
+            ingest_domain_key: domain
+            ingest_snapshot_key: snapshot
+        """
+        super().__init__(params)
+        from data_processing.utils import get_logger
+
+        self.logger = get_logger(__name__)
+
+    def get_transform_config(
```

Comment:
This function looks like a duplicate of that in DataCleaningPythonRuntime.  Can you make a shared method somehow, either as a global or in a shared superclass?

Reply:
Please refer to the discussion in PR https://github.com/IBM/data-prep-kit/pull/691 as to why we implemented like this.